### PR TITLE
[WIP] Add tracing support

### PR DESCRIPTION
**Pull request by freynca**
- Refactoring
- Add zipkin tracing [https://github.com/projectriff/http-gateway/issues/1]

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@freynca Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=projectriff/http-gateway&pullRequestId=20)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=projectriff/http-gateway&pullRequestId=20) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

File: pkg/handlers/message_handler.go [hunk 1]
```diff
@@ -11,8 +11,15 @@ import (
 
 // Creates an http handler that posts the http body as a message to Kafka, replying
 // immediately with a successful http response
-func MessageHandler(producer sarama.AsyncProducer) http.HandlerFunc {
+func MessageHandler(producer sarama.AsyncProducer, traceContext TraceContext) http.HandlerFunc {
```
Comment (ericbottard) [inline pkg/handlers/message_handler.go]: This should be implemented using the Decorator pattern I think

**Review by ericbottard — CHANGES_REQUESTED** [review]
It was hard to review this PR using the GH interface as it touches so many files. A couple of comments though:
- The first 3 commits are actually unrelated and should be part of a different PR. Some things are now exported when then don't deserve to IMO, but let's discuss on a dedicated PR
- the refactoring approach (see above) should go in the direction of making the http-gw an "object" (in the OO sense) to avoid passing so many additional parameters around.
- as to the actual zipkin stuff, I think it would benefit from being implemented as a decorator around a `http.HandlerFunc`

**Comment by ericbottard** [comment]
Also, this will need to be rebased on top of latest master. I'm not sure I understand why so many files were touched in `vendor/`?

**Comment by freynca** [comment]
Closing; will rebase and submit new PRs as needed.

**Pull request closed without merging**